mutters-core/src/main/java/com/rabidgremlin/mutters/core/MatcherScores.java

```diff
@@ -0,0 +1,69 @@
+/* Licensed under Apache-2.0 */
+package com.rabidgremlin.mutters.core;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Holds the matching scores generated by an IntentMatcher.
+ * 
+ * @author rabidgremlin
+ *
+ */
+public class MatcherScores
+{
+  /** Holds the matching scores. */
+  private SortedMap<Double, Set<String>> scores = new TreeMap<Double, Set<String>>();
+
+  /**
+   * Creates a MatcherScores containing no scores.
+   * 
+   */
+  public MatcherScores()
+  {
+    // do nothing
+  }
+
+  /**
+   * Create a MatcherScores with the supplied scores. Map is keyed by score to
+   * allow multiple intents to have the same score.
+   * 
+   * @param scores Map of scores mapped to sets of intent names.
+   */
+  public MatcherScores(SortedMap<Double, Set<String>> scores)
+  {
+    this.scores = scores;
```

Comment:
```this.scores = Objects.requireNonNull(scores);```

can do this in a few places, so it will fail fast on nulls

Reply:
Added to all "core" classes to better enforce usage contracts.